src/components/ArticleInfo.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useEffect, useContext } from "react";
+import { useParams, Link } from "react-router-dom";
+import { getArticleById } from "../api";
+import { Loading } from "./Loading";
+import { LoadingContext } from "../context/LoadingProvider";
+
+export const ArticleInfo = () => {
+  const { article_id } = useParams()
+  const [article, setArticle] = useState(null);
+  const {loading, setLoading} = useContext(LoadingContext)
+
+  useEffect(() => {
+    getArticleById(article_id)
+      .then((data) => {
+        setArticle(data.article)
+        setLoading(false);
+      })
+      .catch((err) => console.error(err));
+  }, [article_id]);
+
+  if (loading) {
+    return (
+      <Loading />
+    );
+  }
+
+  if (!article) {
+    return <p>Article not found!</p>;
+  }
```

Comment:
I like the idea of this, but this should be using something like an error state instead which may be more useful if we know exactly why it sent back an error